org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/internal/test/fixtures/GradleVersionParameterization.java

```diff
@@ -118,22 +120,16 @@ public GradleDistribution apply(GradleVersion input) {
      * is returned that matches the pattern.
      *
      * @param gradleVersionPattern the pattern of Gradle versions to match, must not be null
+     * @param ignoreJavaConstraint whether to ignore the java version constraint or not
      * @return the matching Gradle versions falling into the configured range, never null
      */
-    ImmutableList<GradleVersion> getGradleVersions(String gradleVersionPattern, boolean negateJavaConstraints) {
+    ImmutableList<GradleVersion> getGradleVersions(String gradleVersionPattern, boolean ignoreJavaConstraint) {
         Preconditions.checkNotNull(gradleVersionPattern);
 
         Predicate<GradleVersion> versionConstraint = GradleVersionConstraints.toPredicate(gradleVersionPattern);
         Predicate<GradleVersion> toolingApiConstraint = GradleVersionConstraints.toPredicate(">=2.9");
```

Comment:
Currently the `toolingApiConstraint` limits version >= 2.9. I tried to change it to 2.6 but the following test cases in Java 8 failed:

- Natures and build commands are updated for Gradle #version
- Custom project output location is not updated for #version

So I didn't change it here thus the versions in unsupported gradle properties file start from 2.9. It would be necessary to check if the version is under the limitation of TAPI before we check the properties file, but I have no idea if we should set the limitation to 2.6 or 2.9.

Reply:
Sorry for the late follow-up. 

The minimum tested Gradle version should be 2.6 to be in sync with [what Gradle supports](https://github.com/gradle/gradle/blob/de63e27780bef54829e2cd0b96dcc6bca599cb59/subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiSpecification.groovy#L65). I don't know why it was set to 2.9 in the past.

I'll take a look at this.

Reply:
The failure is expected: the DSL for custom eclipse projects/natures was added in Gradle 2.9. I've created a PR to adjust the the cross-version test range: https://github.com/eclipse/buildship/pull/1198

Reply:
Get it. I'll rebase this PR once #1198 got merged.

Reply:
I've merged #1198

Reply:
I have pushed https://github.com/eclipse/buildship/pull/1161/commits/c2cba1da69822e5898ed810960b16b3ef15b3370 to extend the matrix to include versions from 2.6